torch/distributed/_sharded_tensor/api.py

```diff
@@ -750,3 +796,61 @@ def _create_tensor_from_params(*size, local_device, tensor_init_params: TensorIn
                           device=local_device, )
     else:
         raise ValueError(f'Unsupported create_op: {tensor_init_params.create_op}')
+
+
+class PartialTensor(object):
+    """
+    PartialTensor is an abstraction to represent Tensors that need
+    aggregation across multiple devices and multiple processes.
```

Comment:
We should mention that the PartialTensor `shard` on each rank is of the same size and give a few examples of aggregation functions like `add` to combine all shards of a PartialTensor.